src/Compilers/CSharp/Portable/BoundTree/NullabilityRewriter.cs

```diff
@@ -8,5 +13,68 @@ protected override BoundExpression VisitExpressionWithoutStackGuard(BoundExpress
         {
             return (BoundExpression)Visit(node);
         }
+
+        public override BoundNode VisitBinaryOperator(BoundBinaryOperator node)
+        {
+            return VisitBinaryOperatorBase(node, (binary, left, right, type) => node.Update(node.OperatorKind, node.ConstantValueOpt, node.MethodOpt, node.ResultKind, left, right, type));
```

Comment:
>(binary, left, right, type) => node.Update(node.OperatorKind, node.ConstantValueOpt, node.MethodOpt, node.ResultKind, left, right, type) [](start = 49, length = 136)

This feels wrong. Instead of updating the ```binary``` the lambda will keep updating the original root node over and over again. So, none of the BoundBinaryOperatorBase children collected into the stack below are going to be updated properly. Also, given that we are supposed to be able to update ```BoundBinaryOperator``` and ```BoundUserDefinedConditionalLogicalOperator``` nodes, we cannot rely on the same ```Update``` method to do that. Since none of the issues were caught by the testing, it feels like we are lacking adequate test coverage for the new code in this file.